test/parallel/test-http2-invalidheaderfield.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto) { common.skip('missing crypto'); }
+
+// Check for:
+// Spaced headers
+// Psuedo headers
+// Capitalized headers
+
+const http2 = require('http2');
+const { throws, strictEqual } = require('assert');
+
+const server = http2.createServer(common.mustCall((req, res) => {
+  throws(() => {
+    res.setHeader(':path', '/');
+  }, {
+    code: 'ERR_HTTP2_PSEUDOHEADER_NOT_ALLOWED'
+  });
+  throws(() => {
+    res.setHeader('t est', 123);
+  }, {
+    code: 'ERR_INVALID_HTTP_TOKEN'
+  });
+  res.setHeader('TEST', 123);
+  res.setHeader('test_', 123);
+  res.setHeader(' test', 123);
+  res.end();
+}));
+
+server.listen(0, common.mustCall(() => {
+  const session1 = http2.connect(`http://localhost:${server.address().port}`);
+  session1.request({ 'test_': 123, 'TEST': 123 })
+    .on('end', common.mustCall(() => {
+      session1.close();
+      server.close();
+    }));
+
+  const session2 = http2.connect(`http://localhost:${server.address().port}`);
+  session2.on('error', common.mustCall((e) => {
+    strictEqual(e.code, 'ERR_INVALID_HTTP_TOKEN');
+  }));
+  throws(() => {
+    session2.request({ 't est': 123 });
+  }, {
+    code: 'ERR_INVALID_HTTP_TOKEN'
+  });
+
+  const session3 = http2.connect(`http://localhost:${server.address().port}`);
+  session3.on('error', common.mustCall((e) => {
+    strictEqual(e.code, 'ERR_INVALID_HTTP_TOKEN');
+  }));
+  throws(() => {
+    session3.request({ ' test': 123 });
+  }, {
+    code: 'ERR_INVALID_HTTP_TOKEN'
+  });
+
+  const session4 = http2.connect(`http://localhost:${server.address().port}`);
+  try {
+    session4.request({ ':test': 123 });
+  } catch (e) {
+    strictEqual(e.code, 'ERR_HTTP2_INVALID_PSEUDOHEADER');
+    session4.destroy();
+  }
```

Comment:
This will just go on and not check anything if the code doesn't throw, why not use `throws` here as well?

Reply:
Not destroying the `session4` session would result in a timeout because unlike other errors, a pseudo header error does not automatically destroy the session, hence the catch.

Reply:
Can't you just
```js
const session4 = http2.connect(`http://localhost:${server.address().port}`);
throws(() => {
  session4.request({ ':test': 123 });
}, {
  code: 'ERR_HTTP2_INVALID_PSEUDOHEADER'
});
session4.destroy();
```
since the throw is synchronous anyway?

Also, why the `.destroy()` and not `.close()`? (btw sessions 2 and 3 are not closed either)

Reply:
Sure. I've made the changes, and `close`d the session. Session 2 and 3, being `ERR_INVALID_HTTP_TOKEN` errors, are `destroy`ed internally, hence there was no need to close them.

https://github.com/nodejs/node/blob/651088c3e6b6399a3e656e397c3845b970ad7903/lib/internal/http2/core.js#L1619